src/components/legend/style.js

```diff
@@ -17,16 +17,38 @@ var Color = require('../color');
 
 var subTypes = require('../../traces/scatter/subtypes');
 var stylePie = require('../../traces/pie/style_one');
+var pieCastOption = require('../../traces/pie/helpers').castOption;
+
+var CST_MARKER_SIZE = 12;
+var CST_LINE_WIDTH = 5;
+var CST_MARKER_LINE_WIDTH = 2;
+var MAX_LINE_WIDTH = 10;
+var MAX_MARKER_LINE_WIDTH = 5;
```

Comment:
Please see `legend-constant-itemsizing.png` below and let me know if we can set better constant values.

Reply:
cc @nicolaskruchten @alexcjohnson

Reply:
do these differ from the current defaults?

Reply:
Which _default_ are you referring to here?

Legend items at the moment don't have static defaults, their style depend on the style of their corresponding traces.

Reply:
so I was assuming that constant sizing meant literally "whatever the size would have been if `marker.size` wasn't variable" so I'm confused about what these constants here do

Reply:
You mean what's the default `marker.size` value for a scatter trace?

https://github.com/plotly/plotly.js/blob/5d69e77fde779dc57861e108de720b08a14a3fc4/src/traces/scatter/attributes.js#L365

and it looks like:

![image](https://user-images.githubusercontent.com/6675409/55650510-e5f03d00-57b3-11e9-8517-d190426ac896.png)

I'd argue that's too small when `marker.line.width>0` so I made it twice as big.

Reply:
Ping. Is everyone ok with values I picked for `legend.itemsizing: 'constant'`?

Reply:
> Is everyone ok with values I picked for `legend.itemsizing: 'constant'`?

@etpinard sorry for the late reply. I like how the mock currently looks. However, I have 2 questions:
1. Did you consider sizing it proportionally to the `legend.font.size`? Right now, let's say if `legend.font.size = 30`, the items may look a bit small:
![Screenshot_2019-04-08_18-44-32](https://user-images.githubusercontent.com/301546/55761760-7043d580-5a2e-11e9-997a-1500ca28a699.png)

2. Do we plan to support an attribute to control the size of the items/symbols? If so, would it make more sense to have an `item.(property1|property2|...)` container?

Reply:
> 1. Did you consider sizing it proportionally to the `legend.font.size`?

no



> 2\. Do we plan to support an attribute to control the size of the items/symbols? If so, would it make more sense to have an `item.(property1|property2|...)` container?

https://github.com/plotly/plotly.js/issues/3735

Reply:
Well do you think it is a good idea? The ratio of item size vs font size is good in the current mock. Maybe we should :lock: it down. That being said, I'm open to leaving it as it is since we're planning to implement #3735.